### PR TITLE
some extra tests for queen-attack

### PR DESCRIPTION
**Pull request by IvanRossi**
IMHO, tests should also check for both "positions within board" and "positions are different" inside the attack predicate.

### CODE REVIEW

**Comment by neenjaw** [comment]
I like this idea, thanks for the PR,

I haven't checked, but could you see if these tests are already listed in the `exercism/problem-specifications` repo for this problem?

This is important for a few reasons, in the upcoming version of elixir, the problem introduction will come from that repo. So it is important that our introduction specifies the assumptions present in the exercise.

Also if similar tests exist, could you align them to match their input / expected output?

If there are no similar tests, I think it's reasonable to just add these.

Does that sound reasonable?

File: exercises/practice/queen-attack/.meta/queen_attack.example [hunk 1]
```diff
@@ -1,12 +1,21 @@
-create((X,Y)) :- X >= 0, Y >= 0, 7 >= X, 7 >= Y.
+% Please visit https://exercism.io/tracks/prolog/installation
+% for instructions on setting up prolog.
+% Visit https://exercism.io/tracks/prolog/tests
+% for help running the tests for prolog exercises.
 
-column((_, Y), (_, V)) :- Y =:= V.
+% Replace the goal below with
+% your implementation.
```
Comment (neenjaw) [inline exercises/practice/queen-attack/.meta/queen_attack.example:7]: We don't need this comment in the example file, it's okay to remove.

**Review by neenjaw — CHANGES_REQUESTED** [review]

File: exercises/practice/queen-attack/.meta/queen_attack.example (hunk 1 above)
Comment (IvanRossi) [reply exercises/practice/queen-attack/.meta/queen_attack.example]: > We don't need this comment in the example file, it's okay to remove.

removed and committed.

**Comment by IvanRossi** [comment]
> I haven't checked, but could you see if these tests are already listed in the `exercism/problem-specifications` repo for this problem?
>
> This is important for a few reasons, in the upcoming version of exercism, the problem introduction will come from that repo. So it is important that our introduction specifies the assumptions present in the exercise.
> 
> Also if similar tests exist, could you align them to match their input / expected output?
> 
> If there are no similar tests, I think it's reasonable to just add these.
> 
> Does that sound reasonable?

There is similar stuff but not exactly what i did. For example the "positions within board" are applied just to the "create" property and not the "attack" one, which may make sense in some context (create a queen "object" that you pass to an "attack" function. Current tests look coherent (to me) with what it is now inside `exercism/problem-specifications`.

My 0.02: I would rather have this test applied to the "attack" property or you might circumvent the tests (at least in prolog).

About the test for "queen positions are different" ,  it says within canonical-data.json  comment section:

"Some languages implement tests beyond this set, such as checking, for two pieces being placed on the same position, representing the board graphically, or using standard chess notation. Those tests can be offered as extra credit"

No idea what it is sensible to do now.

**Comment by ErikSchierboom** [comment]
@IvanRossi I think it would be fine to add these tests, as nothing in the instructions contradicts it. Could you resolve the merge conflict?

**Comment by ErikSchierboom** [comment]
@IvanRossi Did you mean to close this PR?

**Pull request closed without merging**